main.js

```diff
@@ -0,0 +1,39 @@
+var $productsContainer = document.querySelector('.products-container');
+var $product = document.createElement('div');
+var $productImage = document.createElement('img');
+var $productInformation = document.createElement('div');
+var $productName = document.createElement('div');
+var $icons = document.createElement('div');
+var $heartIcon = document.createElement('i');
+var $cartPlus = document.createElement('i');
```

Comment:
I totally get the logic when it comes to creating a node and then cloning it for your purposes. But this hurts the level of abstraction you are trying to give to yourself.  Think of it this way. Check next comment